src/bluesky/protocols.py

```diff
@@ -24,10 +25,10 @@
 # Squashes warning
 Dtype = Dtype  # type: ignore
 
-try:
+if sys.version_info >= (3, 10):
     from typing import ParamSpec
-except ImportError:
-    from typing_extensions import ParamSpec  # type: ignore
+else:
+    from typing_extensions import ParamSpec
```

Comment:
It seems that ParamSpec is also available in typing_extensions in later python versions (I tested 3.11). Can't we just import it from there for all versions?

Reply:
I think it makes sense to keep the conditional to avoid installing typing_extensions when not necessary. But I made a mistake since I removed the upper bound. Not sure what the upper bound would be now

To be completely exhaustive also `TypeIs` should be wrapped in an `if sys.version < 3.13`, but at the moment Bluesky doesn't support that version.

Reply:
> I think it makes sense to keep the conditional to avoid installing typing_extensions when not necessary.

But `typing-extensions>=4.10.0` is a production requirement anyway, we're also importing from it for the other python versions [here](https://github.com/jacopoabramo/bluesky/blob/6f51ce3f4bb3c17fb7a8d06b49f41ae8bce3b9c1/src/bluesky/protocols.py#L23)?

> but at the moment Bluesky doesn't support that version.

Yup, no need for that now.

Reply:
Fair enough, marking as resolved with last commit.